lib/matplotlib/image.py

```diff
@@ -978,11 +978,8 @@ def set_extent(self, extent, **kwargs):
             [("x", [extent[0], extent[1]]),
              ("y", [extent[2], extent[3]])],
             kwargs)
-        if len(kwargs):
-            raise ValueError(
-                "set_extent did not consume all of the kwargs passed." +
-                f"{list(kwargs)!r} were unused"
-            )
+        if kwargs:
+            raise _api.kwarg_error("set_extent", next(iter(kwargs)))
```

Comment:
Can we throw this `next(iter(...))` into `kwarg_error` with `if not isinstance(kw, str)`?